### ISSUE TITLE
implicitdet fails for sparse matrices (bug)

### ISSUE THREAD

**Issue by jarlebring** [nep-pack/NonlinearEigenproblems.jl#161]
The implicitdet method does not work for quite simple cases:
```julia
julia> nep=nep_gallery("schrodinger_movebc",10);
julia> implicitdet(nep,λ=0.1,v=ones(10))
```
or
```julia
julia> nep=nep_gallery("dep0_sparse",10);
julia> implicitdet(nep,λ=0.1,v=ones(10))
ERROR: MethodError: no method matching iterate(::SuiteSparse.UMFPACK.UmfpackLU{Complex{Float64},Int64})
Closest candidates are:
  iterate(::Core.SimpleVector) at essentials.jl:589
  iterate(::Core.SimpleVector, ::Any) at essentials.jl:589
  iterate(::ExponentialBackOff) at error.jl:171
  ...
Stacktrace:
 [1] indexed_iterate(::SuiteSparse.UMFPACK.UmfpackLU{Complex{Float64},Int64}, ::Int64) at ./tuple.jl:63
 [2] #implicitdet#14(::Float64, ::Int64, ::Float64, ::Array{Float64,1}, ::Array{Float64,1}, ::Int64, ::Function, ::Type{Complex{Float64}}, ::DEP{Float64,SparseArrays.SparseMatrixCSC{Float64,Int64}}) at /home/jarl/jobb_synced/src/neppack_master/NonlinearEigenproblems.jl/src/method_newton.jl:602
 [3] #implicitdet at ./none:0 [inlined]
 [4] #implicitdet#13 at /home/jarl/jobb_synced/src/neppack_master/NonlinearEigenproblems.jl/src/method_newton.jl:577 [inlined]
 [5] (::getfield(NonlinearEigenproblems.NEPSolver, Symbol("#kw##implicitdet")))(::NamedTuple{(:λ, :v),Tuple{Float64,Array{Float64,1}}}, ::typeof(implicitdet), ::DEP{Float64,SparseArrays.SparseMatrixCSC{Float64,Int64}}) at ./none:0
 [6] top-level scope at none:0
```
Problems with sparse matrices?

**Comment by upadpup** [comment]
Thanks for pointing this out. Seems like there is a problem in general with lu() when applied to sparse matrices in Julia.

`julia> AA = SparseMatrixCSC(rand(Float64,5,5));

julia> lu(AA)
UMFPACK LU Factorization of a (5, 5) sparse matrix
Ptr{Nothing} @0x0000000002c881f0
`
This should have been fixed according to https://github.com/JuliaLang/julia/issues/27387. Fixes from me coming soon.

**Comment by maxbennedich** [comment]
I don't think that's the problem; that issue seemed to have to do with missing imports, resulting in `lu` throwing an error. As far as I can tell, `lu` now works for sparse matrices (although has a confusing `show` message), at least in this example:
```julia
julia> A = sprand(1000, 1000, 0.5);

julia> x = rand(1000);

julia> Alu = lu(A);

julia> @btime $A \ $x;
  130.240 ms (68 allocations: 61.56 MiB)

julia> @btime $Alu \ $x;
  4.229 ms (4 allocations: 62.78 KiB)

julia> A \ x == Alu \ x
true
```

**Comment by upadpup** [comment]
@maxbennedich Indeed. I was confused by the "Ptr{Nothing} @0x0000000002c881f0" in the message. But I have fixed the problem now. Calling lu() with three arguements like
`L,U,P = lu(AA)`
throws an error for sparse matrices. The correct way is to just use 
`F = lu(AA)`
and use backslash with the F, as you showed. I have fixed this problem in implicitdet() now:
```
julia> nep = nep_gallery("dep0_sparse",10);

julia> λ,v = implicitdet(nep);

julia> compute_resnorm(nep,λ,v)
2.424148044738925e-16
```
